### DISCUSSION TITLE
Reading node position, size and colour from files by updating the Recast Column plugin

### DISCUSSION THREAD

**Discussion by TheG-Meister** [gephi/gephi#2771] [Q&A] [answered]
Hi! I'm working on a project that could really make use of importing node position and colour from external files. I've seen that gephi can export said columns but I can't read them back in using .tsv format. I've heard the [Recast Column](https://gephi.org/plugins/#/plugin/recast-column) plugin can let you set these properties after import, but it seems like it's last release was for version 0.8.2, over a decade ago. Having contacted the original creator they said they didn't want to update it themselves, so I was thinking about remaking it myself.

The original code base for version 0.8.2 can be found [here](https://github.com/tarasklas/gephi/tree/change-type-of-column), and from having a quick look it seems like the majority of the plugin's additional code can be found in a single module folder with minimal changes to files in other locations. Would it be possible to turn this into a new plugin or would a complete rebuild be required? Any advice would be appreciated!

Many thanks, G

**Comment by mbastian** [comment, marked as answer]
Hi @TheG-Meister, seems like indeed we have a few options. 

1.  If the TSV import doesn't read positions and color it could be added. Feel free to create an Issue for that. It can be fixed.
2. This plugin's code would need to be migrated to 0.9+, which is a bit of work. Alternatively, you can [start from scratch](https://github.com/gephi/gephi-plugins#get-started) and copy-paste the few classes as it's not too much. Recreating from scratch is probably faster than migrating.

Finally, as a workaround picking another file format, like GDF for instance.

**Reply by TheG-Meister** [reply]
Thanks for the tips! These seem great, I'll follow up on both of them.
